# Gender-based methylation differences

```{r setup, include=FALSE}
# source("functions/knit_wiki.r")
# knit_wiki("gender-based-differential-meth-report.Rmd", output.filename = "Gender-based methylation differences")

report.title <- "gender-based-meth-differences"
fig.dir <- file.path("figures", report.title)
fig.prefix <- "meth-gender-"

dir.create(fig.dir, showWarnings = FALSE)

opts_knit$set(upload.fun = function(file, 
  key = "ec071b3a0c69857c0db26a477c1a568c") imgur_upload(file, key))

opts_chunk$set(fig.path = file.path(fig.dir, fig.prefix),
  cache.path = paste(".cache-", report.title, "/", sep = ""),
  echo=FALSE, results = 'hide',
  cache=FALSE, fig.width=5, fig.height=4, dpi=150)
```

```{r load-libraries}
library(plyr)
library(reshape2)
library(ggplot2)

# ggpairs function
devtools::source_gist("3298388")

source("functions/heat_matrix.r")
source("functions/print_xtable.r")
source("functions/save-figures.r")
source("functions/plot_probe.r")
```

```{r load-data, cache=FALSE}
source("4b-load-meth-data.r")

t.results <- read.csv("results/gender-based-differential-methylation.csv",
                      stringsAsFactors = FALSE)
```

Explore and visualize results from the gender-based differential methylation analysis, which sought to identify loci with significantly different levels of methylation between male and female samples.

* script: [`gender-based-differential-meth.r`](https://github.com/aaronwolen/gibbs-brain-project/blob/master/gender-based-differential-meth.r)
* results: [`gender-based-differential-methylation.csv`](https://github.com/aaronwolen/gibbs-brain-project/blob/master/results/gender-based-differential-methylation.csv)

## Methods
The analysis was restricted to only autosomal probes. Low-variance probes were filtered out by removing any probes with an experiment-wide standard deviation < 0.1. 

Methylation levels were adjusted for age and sample tissue bank by fitting a linear model with these variables and using the residuals in subsequent analyses. These models were fit independently for each brain region so the residuals produced would be approximately normal within each region. 

The t.test was carried out using the `rowttests` from the `genefilter` package. Since `rowttests` is unable to handle missing data, any probes with NA values had to be tested individually after removing offending samples.

Multiple testing corrections were performed within each region using the Benjamini & Hochberg method for generating q-values.

## Results

```{r results-table, results='asis'}
results.table <- ddply(t.results, .(tissue), summarise, 
  'p-value < 0.05' = sum(p.value < 0.05),
  'q-value < 0.1' = sum(q.value < 0.1))

print_xtable(results.table)
```

### Volcano plot
Volcano plot of mean gender-based methylation differences plotted against raw t-test p-values. The outlier probe cg15915418 was removed prior to plotting. 

```{r volcano-plot}
vp <- ggplot(t.results) +
  aes(x = dm, y = -log10(p.value)) +
  geom_point(size = 1) +
  facet_wrap(~tissue) +
  xlab("Mean gender difference") + 
  ylab(expression(-log[10] * " p-value"))

# Remove TLE1 outlier
vp %+% subset(t.results, id != "cg15915418")
```


### Pairwise tissue comparisons of mean gender differences
Pairwise comparison of gender-based methylation differences across all tissues. As above, cg15915418 was removed prior to plotting. 

```{r dm-tissue-pairs, fig.width=5.5}
tissues <- levels(eset.meth$tissue)

t.dm <- dcast(t.results, id ~ tissue, value.var = "dm")

ggpairs(data = subset(t.dm, id != "cg15915418"), facet.vars = tissues, 
  facet.scale = "fixed", alpha = 0.5, size = 0.75) + 
  stat_smooth(method = "lm") + coord_fixed() +
  xlab("Mean gender difference")  + ylab("Mean gender difference")
  opts(axis.text.x = theme_text(angle = 90, hjust = 1, 
    colour = "grey50", size = base_size * 0.8)) 
```

### Mean difference correlations across tissues

As seen in the pairs plot above, gender-based methylation differences do correlate across tissues to some extent. The following heatmap depicts the inter-tissue correlations, which were strongest between PONS/FCTX and TCTX/FCTX. Re-calculating these correlations using only probes with t-test p-values < 0.05 increased these correlations by 0.15 - 0.22. 

```{r dm-tissue-correlations, fig.width=3.5, fig.height=2.5}

dm.cors <- cor(t.dm[, tissues])
heat_matrix(dm.cors, symmetric = TRUE, legend.title = "cor", font.size = 3)

# Limit correlations to probes with larger gender differences
sig.probes <- unique(subset(t.results, p.value < 0.05)$id)
sig.dm.cors <- cor(subset(t.dm, id %in% sig.probes)[, tissues])

sig.dm.cors - dm.cors
```

### What's up with cg15915418?

As shown in the table below, cg15915418's differential methylation results clearly make it an outlier, especially when compared to the range of p-values for all other probes (voclano plot, above).
```{r results-cg15915418, results='asis'}
print_xtable(subset(t.results, id == "cg15915418"),
  display = c("p.value" = "e", "q.value" = "e"))
```

The following barplot verifies there is indeed a huge and consistent difference in cg1591541 methylation betweengenders, with females showing almost complete repression in all four tissues. As this pattern is characteristic of probes targeting the sex chromosomes, I looked into the possibility that cg15915418 is mis-annotated.  
```{r barplot-cg15915418, fig.width = 7.5}

probe_barplot(eset.meth, probe = "cg15915418", fill = "gender", 
              facet.row = "tissue", base.size = 8)

save_plot("barplot-cg15915418.pdf", width = 7.5, height = 4, path = fig.dir)
```

Blating cg15915418's `sourceseq` against version hg19 of the human regnome revealed a strong match on the X chromosome ([full results](http://genome.ucsc.edu/cgi-bin/hgc?o=64627510&g=htcUserAli&i=../trash/hgSs/hgSs_genome_47ba_dabb20.pslx+..%2Ftrash%2FhgSs%2FhgSs_genome_47ba_dabb20.fa+YourSeq&c=chrX&l=64627510&r=64627560&db=hg19&hgsid=284541277)), in addition to its purported Chr 9 target. As such, it's likely cg15915418 is actually measuring X chromosome methylation and should be re-annotated. 

The barplot above also reveals a potential sample mix-up as individual 1886's methylation levels are on par with other males in cerebellum only. Genereating this plot for a random sample of other X chr probes showing sex-specific methylation patterns also suggested the sample labeled as 1886 cerebellum comes from a male. 

```{r examine-sig-probes, eval=FALSE}

# Produce visualizations for individiual probes with largest gender differences
sig.results <- subset(t.results, q.value < .1 & id != "cg15915418")
sig.results <- arrange(sig.results, q.value)

top.probes <- unique(sig.results$id)[1:20]

# Barplot providing each individual's tissue-specific methylation level
pdf(file.path(fig.dir, paste(fig.prefix, "top-probes-barplot.pdf", sep = "")), 
  width = 10, height = 6)
for(p in top.probes) {
  plot(probe_barplot(eset.meth, probe = p, fill = "gender", base.size = 8))
}
dev.off()

# Dotplot of methylation levels with mean and sd summarizations
pdf(file.path(fig.dir, paste(fig.prefix, "top-probes-dotplot.pdf", sep = "")),
  width = 10, height = 6)
for(p in top.probes) {
  plot(probe_dotplot(eset.meth, probe = p, color = "age", base.size = 12) +
    stat_summary(fun.data = function(x) mean_sdl(x, mult = 1), 
      color = "grey50", geom = "crossbar", alpha = 0.2))
}
dev.off()
```